fully-client-side-calculator/index.js

```diff
@@ -0,0 +1,124 @@
+const screen = document.getElementById("screen");
+const numbers = document.querySelectorAll(".numbers-to-press");
+const operationButtons = document.querySelectorAll(".operation-buttons");
+const equalsButton = document.getElementById("equalsButton");
+const clearButton = document.getElementById("clearButton");
+const decimalButton = document.getElementById("decimalButton");
+let secondValue = "";
+let operand = "";
+let currentValue = 0;
+let operantPressed = false;
+let timesOperandUsed = 0;
+let equalsButtonPurposelyPressed = false;
+let nothingHappened = true;
+const calculator = new Calculator();
+
+function returnValueOfOperation() {
+  if (screen.value === "" || screen.value === ".") return;
+  currentValue = parseFloat(secondValue);
+
+  switch (operand) {
+    case "+":
+      calculator.add(currentValue);
+      break;
+    case "-":
+      calculator.subtract(currentValue);
+      break;
+    case "*":
+      calculator.multiply(currentValue);
+      break;
+    case "/":
+      calculator.divide(currentValue);
+      break;
+    case "^":
+      calculator.power(currentValue);
+      break;
+  }
+
+  screen.value = calculator.getTotal();
+}
+
+clearButton.addEventListener("click", () => {
+  screen.value = "0";
+  secondValue = "";
+  timesOperandUsed = 0;
+  operantPressed = false;
+  equalsButtonPurposelyPressed = false;
+  nothingHappened = true;
+  calculator.clear();
+});
+
+numbers.forEach((button) => {
+  button.addEventListener("click", () => {
+    if (screen.value === "0" && operantPressed === false) {
+      screen.value = "";
+    }
+    // After you purposly press equalls button, you get an answer but then you can change that answer
+    if (equalsButtonPurposelyPressed) {
+      screen.value = "";
+      secondValue = "";
+    }
+
+    // if when second number entered is at 0., then number button is pressed, we dont want screen value to be reset,
+    //  we want the number button pressed to be concatnated to the 0.
+    if (operantPressed && screen.value === "0.") {
```

Comment:
I think if the screen value is 0., then operandPressed will never be true, so this section of code can never run and is redundant